nomad/stream/subscription.go

```diff
@@ -124,68 +124,55 @@ func (s *Subscription) Unsubscribe() {
 // filter events to only those that match a subscriptions topic/keys/namespace
 func filter(req *SubscribeRequest, events []structs.Event) []structs.Event {
 	if len(events) == 0 {
-		return events
+		return nil
 	}
 
-	var count int
-	for _, e := range events {
-		_, allTopics := req.Topics[structs.TopicAll]
-		if _, ok := req.Topics[e.Topic]; ok || allTopics {
-			var keys []string
-			if allTopics {
-				keys = req.Topics[structs.TopicAll]
-			} else {
-				keys = req.Topics[e.Topic]
-			}
-			if req.Namespace != "" && e.Namespace != "" && e.Namespace != req.Namespace {
-				continue
-			}
-			for _, k := range keys {
-				if e.Key == k || k == string(structs.TopicAll) || filterKeyContains(e.FilterKeys, k) {
-					count++
-				}
-			}
+	allTopicKeys := req.Topics[structs.TopicAll]
+
+	var result []structs.Event
```

Comment:
I think that it may be worth checking if we can send the events unfiltered as a small optimization, thoughts?

```go

if req.Namespace == "" && len(allTopicKeys) == 1 && allTopicKeys[0] == "*" {
return events
}
```